modules/features2d/src/hal_replacement.hpp

```diff
@@ -101,8 +101,9 @@ inline int hal_ni_FAST_NMS(const uchar* src_data, size_t src_step, uchar* dst_da
    @param threshold Threshold for keypoint
    @param nonmax_suppression Indicates if make nonmaxima suppression or not.
    @param type FAST type
+   @param realloc_func function for reallocation
 */
-inline int hal_ni_FAST(const uchar* src_data, size_t src_step, int width, int height, uchar* keypoints_data, size_t* keypoints_count, int threshold, bool nonmax_suppression, int /*cv::FastFeatureDetector::DetectorType*/ type) { return CV_HAL_ERROR_NOT_IMPLEMENTED; }
+inline int hal_ni_FAST(const uchar* src_data, size_t src_step, int width, int height, void** keypoints_data, size_t* keypoints_count, int threshold, bool nonmax_suppression, int /*cv::FastFeatureDetector::DetectorType*/ type, void* (*realloc_func)(void*, size_t)) { return CV_HAL_ERROR_NOT_IMPLEMENTED; }
```

Comment:
We usually presume backward compatibility in HAL. I propose to presume the original API and add new entrypoint with realloc, e.g. `hal_ni_FAST_withRealloc`.
On caller side you need to try the new one, then the old one and then default impl.